### ISSUE TITLE
about the Depth-Guided Refinement Stage

### ISSUE THREAD

**Issue by hello-trouble** [cxtalk/PDR-GAN#1]
您好，谢谢您的分享。打扰您了，有点困惑，请教一下，（1）在the Depth-Guided Refinement Stage对应的代码中，鉴别器中判断为真的是：
 depth_real, pred_real = self.netStage3_DepthGuidedD(self.stage3_ref_J)
 loss_D_real = self.stage2_criterionGAN(pred_real, True)
判断为假对应的代码为：
fake_J = self.fake_J_pool.query(self.stage2_ref_J)
depth_fake,pred_fake = self.netStage2_DepthGuidedD(fake_J.detach())
loss_D_fake = self.stage2_criterionGAN(pred_fake, False)
真数据“self.stage3_ref_J” 和假数据“self.stage2_ref_J” 都是生成器生成的数据，这样的对抗似乎逻辑上有点难以理解？而在HeavyRain 论文及他们后来的去雾文章中是用配对的图像来训练的，有对应的GT图像指导。恳请指点 （2）另外不知是否方便提供一下训练的命令和训练好的模型呢？再次感谢您的分享。

**Comment by wrl517** [comment]
> 



> 您好，谢谢您的论文和分享。打扰您了，有点困惑，想请教一下，（1）在the Depth-Guided Refinement Stage对应的代码中，鉴别器中判断为真的代码是： depth_real, pred_real = self.netStage3_DepthGuidedD(self.stage3_ref_J) loss_D_real = self.stage2_criterionGAN(pred_real, True) 判断为假对应的代码为： fake_J = self.fake_J_pool.query(self.stage2_ref_J) depth_fake,pred_fake = self.netStage2_DepthGuidedD(fake_J.detach()) loss_D_fake = self.stage2_criterionGAN(pred_fake, False) 真数据“self.stage3_ref_J” 和假数据“self.stage2_ref_J” 都是生成器生成的数据，这样的对抗似乎逻辑上感觉不是很好理解？在HeavyRain 论文及他们后来的去雾文章中是用配对的图像来训练的，有对应的GT图像指导，咱们这里是内容不同的生成数据，通常情况都是当作“假”数据来处理的，不知道我哪里理解的不正确，恳请指点解惑 （2）另外不知是否方便提供一下训练的命令和训练好的模型呢？祝您工作愉快，学习进步，再次感谢您的分享。

您好，打扰您了，请问您运行出来了吗，最近刚接触这个模型一直无法顺利运行，能否请教一下您？
